src/fluidsynth.c

```diff
@@ -954,10 +963,13 @@ int main(int argc, char **argv)
             fluid_server_join(server);
         }
 
+#ifdef SYSTEMD_SUPPORT
+        sd_notify(0, "STOPPING=1");
```

Comment:
Shouldn't this `STOPPING` notification happen before `fluid_server_join()`?

Reply:
I've tried that and it doesn't work because the service would be in stopping state soon after it is started and systemd would ultimately kill it.

STOPPING=1 is used to tell the service manager that the service is beginning its shutdown.
fluid_server_join is a blocking call that waits for the server thread to quit.
systemd stops the fluidsynth process by sending a SIGTERM signal that causes the server thread to exit, which in turn makes the call to fluid_server_join return.
So from systemd's perspective the shutdown sequence starts _after_ the call to fluid_server_join.